src/gui_common/ModalManager.cs

```diff
@@ -0,0 +1,189 @@
+﻿using System.Collections.Generic;
+using Godot;
+using Godot.Collections;
+
+/// <summary>
+///   Handles a stack of <see cref="CustomPopup"/>s that blocks GUI inputs.
```

Comment:
Don't we have a ton of popup derived things that can be left open while other stuff is done? I feel like this and some other of the comments in this PR seem to imply that, that very common use case would no longer work.

Reply:
They still work, they're not mentioned anywhere here because they don't need modality. For non-modal popups, setting them just as top-level with `SetAsTopLevel` is sufficient.